src/components/dialog-full-screen/dialog-full-screen.pw.tsx

```diff
@@ -0,0 +1,565 @@
+    await dialogNested.click();
+    await page.keyboard.press("Tab");
+    const closeIcon = getDataElementByValue(page, "close").nth(2);
+    await expect(closeIcon).toHaveCSS(
+      "box-shadow",
+      "rgb(255, 188, 25) 0px 0px 0px 3px, rgba(0, 0, 0, 0.9) 0px 0px 0px 6px"
+    );
+    await expect(closeIcon).toHaveCSS("outline", "rgba(0, 0, 0, 0) solid 3px");
+  });
+
+  test("should render nested dialogs with the aria-modal property only set on the top one, even when the dialogs are wrapped in separate CarbonProviders", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<MultipleDialogsInDifferentProviders />);
+
+    await page.getByRole("button").filter({ hasText: `Open Modal 1` }).click();
+    const fullDialog = getDataElementByValue(page, "dialog-full-screen");
+    await expect(fullDialog).toHaveAttribute("aria-modal", "true");
+
+    await page.getByRole("button").filter({ hasText: `Open Modal 2` }).click();
+    await expect(fullDialog).not.toHaveAttribute("aria-modal", "true");
+    await expect(page.getByRole("dialog")).toHaveAttribute(
+      "aria-modal",
+      "true"
+    );
+  });
+
+  test("should render DialogFullScreen component with aria-describedby", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<DialogFullScreenComponent aria-describedby={testAria} />);
+
+    await expect(page.getByRole("dialog")).toHaveAttribute(
+      "aria-describedby",
+      testAria
+    );
+  });
+
+  test("should render DialogFullScreen component with aria-label", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<DialogFullScreenComponent aria-label={testAria} />);
+
+    await expect(page.getByRole("dialog")).toHaveAttribute(
+      "aria-label",
+      testAria
+    );
+  });
+
+  test("should render DialogFullScreen component with aria-labelledby", async ({
+    mount,
+    page,
+  }) => {
+    await mount(
+      <DialogFullScreenWithTitleAsReactComponent aria-labelledby={testAria} />
+    );
+
+    await expect(page.getByRole("dialog")).toHaveAttribute(
+      "aria-labelledby",
+      testAria
+    );
+  });
+
+  test("should render DialogFullScreen component using focusFirstElement", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<DialogFullScreenComponent />);
+
+    const focusedButton = page
+      .getByRole("button")
+      .filter({ hasText: "This should be focused first now" });
+    await expect(focusedButton).toHaveCSS(
+      "box-shadow",
+      "rgb(255, 188, 25) 0px 0px 0px 3px, rgba(0, 0, 0, 0.9) 0px 0px 0px 6px"
+    );
+    await expect(focusedButton).toHaveCSS(
+      "outline",
+      "rgba(0, 0, 0, 0) solid 3px"
+    );
+  });
+
+  test("should render Dialog component disabling autofocus", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<DialogFullScreenComponent disableAutoFocus />);
+
+    const focusedButton = page
+      .getByRole("button")
+      .filter({ hasText: "This should be focused first now" });
+    await expect(focusedButton).not.toHaveCSS(
```

Comment:
I think this test is a bit too general, not.toHaveCSS could make a lot of values pass as long as it's not  "box-shadow", "rgb(255, 188, 25) 0px 0px 0px 3px, rgba(0, 0, 0, 0.9) 0px 0px 0px 6px". Suggest to change to something like below: **const hasFocus = await focusButton.isFocused();
expect(hasFocus).toBeFalsy();**